src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -614,36 +595,94 @@ describe("Decimal", () => {
     );
 
     describe("i18n", () => {
-      beforeAll(() => {
-        I18n.locale = "fr";
+      describe("es", () => {
+        const esProps = { locale: "es-ES" };
+
+        it.each([
+          ["1000.23", "1000,23"],
+          ["10000.00", "10.000,00"],
+          ["10000.23", "10.000,23"],
+          ["100000.00", "100.000,00"],
+          ["1000000.00", "1.000.000,00"],
+        ])("format %s to %s", (a, b) => {
+          render({ value: a, ...esProps });
+          expect(value()).toBe(b);
+          expect(hiddenValue()).toBe(a);
+        });
       });
 
-      afterAll(() => {
-        I18n.locale = undefined;
+      describe("pt", () => {
+        const ptProps = { locale: "pt-PT" };
```

Comment:
I think it would be great if you could add a unit test that covers `1,1,1,1` not being converted to `1.1.1.1` with the `pt-PT` locale, as per the bug @DlgSHi found. It means when we come to refactor the component to be a functional component in the near future, we have a unit test that already covers this scenario.